cloud/disk_manager/internal/pkg/auth/credentials.go

```diff
@@ -26,9 +28,30 @@ func NewCredentials(
 	if len(config.GetMetadataUrl()) == 0 {
 		return nil
 	}
-
-	return &credentialsWrapper{
-		impl: metadata.NewInstanceServiceAccount(
+	var (
+		impl Credentials
+		err  error
+	)
+	if config.GetServiceAccount() != nil {
+		impl, err = credentials.NewOauth2TokenExchangeCredentials(
+			credentials.WithTokenEndpoint(config.GetMetadataUrl()),
+			credentials.WithAudience(config.GetServiceAccount().GetAudience()),
+			credentials.WithJWTSubjectToken(
+				credentials.WithSigningMethod(jwt.SigningMethodRS256),
+				credentials.WithKeyID(config.GetServiceAccount().GetKeyId()),
+				credentials.WithRSAPrivateKeyPEMFile(config.GetCertFile()),
+				credentials.WithIssuer(config.GetServiceAccount().GetId()),
+				credentials.WithSubject(config.GetServiceAccount().GetId()),
+				credentials.WithAudience(
+					config.GetServiceAccount().GetAudience()),
+			),
+		)
+		if err != nil {
+			logging.Error(ctx, "failed to create token credentials: %s", err)
+			return nil
```

Comment:
Why?
This function should return an error

Reply:
Let us make this function return an error in another PR.